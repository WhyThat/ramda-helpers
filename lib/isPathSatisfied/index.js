import { curry, pathSatisfies } from 'ramda';

import isSet from '../isSet';
import transformer from '../transformDotStringToList';

const isPathDefined = pathSatisfies(isSet);

const isPathSatisfied = curry((predicate, obj) => isPathDefined(transformer(predicate), obj));

/**
 *  Returns `true` if the specified object property at given path isSet @see isSet
 *  returns `false` otherwise
 *  @name isPathSatisfied
 *  @memberof RamdaHelper#
 *  @param {string|string[]} propPath
 *  @param {*} obj
 *  @returns {Boolean}
 *  @example
 *
 *  isPathSatisfied('a.b.c', { a: { b: { c: 'c'} } }) //=> true
 *  isPathSatisfied(['a', 'b', 'c'], { a: { b: { c: 'c'} } }) //=> true
 *  isPathSatisfied('a.c.d', { a: { b: { c: 'c'} } }) //=> false
 *
 */
export default isPathSatisfied;
