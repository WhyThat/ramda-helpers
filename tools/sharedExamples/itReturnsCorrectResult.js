export default (result) => {
  test('returns correct result', () => {
    expect(result).toMatchSnapshot();
  });
};
