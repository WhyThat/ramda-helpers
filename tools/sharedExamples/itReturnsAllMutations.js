import itReturnsCorrectResult from './itReturnsCorrectResult';

export default (fn, mutations) => {
  mutations.forEach((mutation, index) => {
    const name = mutation.name || `mutations ${index}`;

    describe(name, () => {
      itReturnsCorrectResult(fn(mutation.props));
    });
  });
};
