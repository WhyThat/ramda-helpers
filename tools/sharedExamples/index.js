
import itReturnsAllMutations from './itReturnsAllMutations';
import itReturnsCorrectResult from './itReturnsCorrectResult';

export {
  itReturnsAllMutations,
  itReturnsCorrectResult,
};
