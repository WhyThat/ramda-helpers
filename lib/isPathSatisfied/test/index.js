import { itReturnsCorrectResult } from '../../test/sharedExamples';
import isPathSatisfied from '../';

const obj1 = { a: { b: { c: {} } } };

describe('isPathSatisfied', () => {
  const mutations = [{
    name: 'With empty props',
    props: ['', {}],
  }, {
    name: 'With empty string predicate and an object',
    props: ['', obj1],
  }, {
    name: 'With a string predicate and truthy path (deepth == 1)',
    props: ['a', obj1],
  }, {
    name: 'With a string predicate and truthy path (deepth > 1)',
    props: ['a.b', obj1],
  }, {
    name: 'With a string predicate and falsy path (deepth == 1)',
    props: ['b', obj1],
  }, {
    name: 'With a string predicate and falsy path (deepth > 1)',
    props: ['a.c', obj1],
  }, {
    name: 'With a string predicate and falsy path (deepth > 1)',
    props: ['a.c.d', obj1],
  }, {
    name: 'With empty array predicate and an object',
    props: [
      [], obj1,
    ],
  }, {
    name: 'With a array predicate and truthy path (deepth == 1)',
    props: [
      ['a'], obj1,
    ],
  }, {
    name: 'With a array predicate and truthy path (deepth > 1)',
    props: [
      ['a', 'b'], obj1,
    ],
  }, {
    name: 'With a array predicate and falsy path (deepth == 1)',
    props: [
      ['b'], obj1,
    ],
  }, {
    name: 'With a array predicate and falsy path (deepth > 1)',
    props: [
      ['a', 'c'], obj1,
    ],
  }];

  mutations.forEach((mutation) => {
    describe(mutation.name, () => {
      itReturnsCorrectResult(isPathSatisfied(...mutation.props));
    });
  });
});

