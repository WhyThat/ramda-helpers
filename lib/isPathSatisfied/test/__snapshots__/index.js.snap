// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isPathSatisfied With a array predicate and falsy path (deepth == 1) returns correct result 1`] = `false`;

exports[`isPathSatisfied With a array predicate and falsy path (deepth > 1) returns correct result 1`] = `false`;

exports[`isPathSatisfied With a array predicate and truthy path (deepth == 1) returns correct result 1`] = `true`;

exports[`isPathSatisfied With a array predicate and truthy path (deepth > 1) returns correct result 1`] = `true`;

exports[`isPathSatisfied With a string predicate and falsy path (deepth == 1) returns correct result 1`] = `false`;

exports[`isPathSatisfied With a string predicate and falsy path (deepth > 1) returns correct result 1`] = `false`;

exports[`isPathSatisfied With a string predicate and falsy path (deepth > 1) returns correct result 2`] = `false`;

exports[`isPathSatisfied With a string predicate and truthy path (deepth == 1) returns correct result 1`] = `true`;

exports[`isPathSatisfied With a string predicate and truthy path (deepth > 1) returns correct result 1`] = `true`;

exports[`isPathSatisfied With empty array predicate and an object returns correct result 1`] = `false`;

exports[`isPathSatisfied With empty props returns correct result 1`] = `false`;

exports[`isPathSatisfied With empty string predicate and an object returns correct result 1`] = `false`;
